Responsive font-sizes implementation

MartijnCuppens opened the pull request:
Implementation of responsive font-size system [RFS](https://github.com/MartijnCuppens/rfs/).

Link to feature-request:
https://github.com/twbs/bootstrap/issues/23053

CC: @XhmikosR, @kimamil, @mdo

MartijnCuppens committed:
RFS implementation

MartijnCuppens committed:
Add RFS-support

MartijnCuppens committed:
RFS update & RFS configuration added

MartijnCuppens committed:
Add documentation

MartijnCuppens committed:
Responsive fixes

MartijnCuppens committed:
Use px-rendering by default to avoid font-size change when changing the html-font-size

MartijnCuppens committed:
Syntax fixes

MartijnCuppens committed:
Merge branch 'v4-dev' into rfs

mabar commented:
[Poly fluid sizing](https://www.smashingmagazine.com/2017/05/fluid-responsive-typography-css-poly-fluid-sizing/#the-holy-grail-of-css-sizing) would be better. More fluent.

MartijnCuppens commented:
@mabar, this implementation doesn't require to define mappings for every font-size so it can easily be used by users who do not have much frontend experience.

kimamil reviewed (changes requested):
Can you add an example in the documentation on how to use this with rem? As I understood, rfs must be 'false' and unit set to 'rem'. Then I could use media-breakpoints on the html tag to set the font-size in 'rem'? But basically you suggest using fixed px sizes to set the base font-size in all breakpoints, correct?

Else, nice catch!

MartijnCuppens committed:
RFS documentation changes

- Explain configuration
- Remove link to github-repo from configuration
- Remove old implementation of bootstrap

MartijnCuppens committed:
Merge branch 'v4-dev' into rfs

XhmikosR commented on docs/4.0/content/responsive-font-sizes.md:
```diff
@@ -7,48 +7,111 @@ toc: true
 
 ## Responsive font sizes
-Bootstrap uses <a href="https://github.com/MartijnCuppens/rfs" target="_blank" rel="noopener">RFS</a> to control its font-size. RFS is an SCSS-mixin which automatically calculates the correct font size based on the browsers screen width. You just have got to define your font-size for big screens and the font size will automatically decrease for smaller screens.
+Bootstrap uses <a href="https://github.com/MartijnCuppens/rfs" target="_blank" rel="noopener">RFS</a> to control its font size. RFS is an SCSS-mixin which automatically calculates the correct font size based on the browsers screen width. You just have got to define your font size for big screens and the font size will automatically decrease for smaller screens.
```
Please change this to a normal markdown link. Same for other ones too if any.

MartijnCuppens replied:
Done: https://github.com/twbs/bootstrap/pull/23157/commits/aa676531a18d3a099de676e9cfcbdad6fb205ed1

MartijnCuppens commented:
@kimamil, the previous documentation might indeed be a little bit confusing since it didn't explain how the configuration works but explained how you disable RFS and use rem-units for font-resizing just like you can do now.

I've updated the documentation and tried to explain how every setting of the configuration changes the way the responsive font size is calculated. Is this a bit clear or does anyone has any suggestions on how to clarify this even more?

MartijnCuppens committed:
Change html-link to markdown-link

MartijnCuppens committed:
Merge v4-dev in rfs-branch

MartijnCuppens committed:
Merge branch 'v4-dev' into rfs

MartijnCuppens committed:
Revert changes
